mesonbuild/coredata.py

```diff
@@ -746,16 +746,20 @@ def set_default_options(self, default_options, subproject, env):
         # Some options default to environment variables if they are
         # unset, set those now. These will either be overwritten
         # below, or they won't. These should only be set on the first run.
-        if env.first_invocation:
-            p_env = os.environ.get('PKG_CONFIG_PATH')
-            if p_env:
-                # PKG_CONFIG_PATH may contain duplicates, which must be
-                # removed, else a duplicates-in-array-option warning arises.
-                pkg_config_paths = []
-                for k in p_env.split(':'):
-                    if k not in pkg_config_paths:
-                        pkg_config_paths.append(k)
-                options['pkg_config_path'] = pkg_config_paths
+        p_env = os.environ.get('PKG_CONFIG_PATH')
+        if p_env:
+            # PKG_CONFIG_PATH may contain duplicates, which must be
+            # removed, else a duplicates-in-array-option warning arises.
+            p_list = list(OrderedSet(p_env.split(':')))
```

Comment:
not sure if this is relevant here, but Windows uses ";" not ":"

Reply:
We had that conversation, I don't remember where... There are multiple places that does that mistake and fixing it caused issues in the CI that we did not understand. Definitely something that needs to be fixed at some point.